eth/filters/api.go

```diff
@@ -345,6 +347,20 @@ func (api *PublicFilterAPI) GetLogs(ctx context.Context, crit FilterCriteria) ([
 		if crit.ToBlock != nil {
 			end = crit.ToBlock.Int64()
 		}
+
+		beginBlock, err := api.backend.HeaderByNumber(ctx, rpc.BlockNumber(begin))
+		if err != nil {
+			return nil, fmt.Errorf("couldn't find fromBlock, fromBlock: %d", begin)
+		}
+		endBlock, err := api.backend.HeaderByNumber(ctx, rpc.BlockNumber(end))
+		if err != nil {
+			return nil, fmt.Errorf("couldn't find toBlock, toBlock: %d", end)
+		}
+		realBegin := beginBlock.Number.Int64()
+		realEnd := endBlock.Number.Int64()
+		if realEnd-realBegin+1 > api.maxBlockRange {
+			return nil, fmt.Errorf("block range is bigger than maxBlockRange, block range: %d, maxBlockRange: %d", realEnd-realBegin+1, api.maxBlockRange)
+		}
```

Comment:
If the maxBlockRange reached, why not return the begin to maxBlockRang instead of returning the error

Reply:
That is very confusing behaviour, the caller has no way to know that they did not get all the data that they asked for.

Reply:
ok, get it.